features/nanostack/FEATURE_NANOSTACK/targets/TARGET_SL_RAIL/NanostackRfPhyEfr32.cpp

```diff
@@ -152,6 +187,103 @@ static int8_t rf_start_cca(uint8_t *data_ptr, uint16_t data_length, uint8_t tx_h
 /* Local function prototypes */
 static bool rail_checkAndSwitchChannel(uint8_t channel);
 
+static void rf_thread_loop(const void *arg)
+{
+    tr_debug("rf_thread_loop: starting (id: %d)\n", rf_thread_id);
+    for (;;) {
+        osEvent event = osSignalWait(0, osWaitForever);
+
+        if (event.status != osEventSignal) {
+            continue;
+        }
+
+        platform_enter_critical();
+
+        if (event.value.signals & SL_RX_DONE) {
+            tr_debug("rf_thread_loop: SL_RX_DONE signal received\n");
+
+            while(rx_queue_tail != rx_queue_head) {
+                void* handle = (void*) rx_queue[rx_queue_tail];
+
+                RAIL_RxPacketInfo_t* info = (RAIL_RxPacketInfo_t*) memoryPtrFromHandle(handle);
+                tr_debug("rf_thread_loop: rx %d bytes with rssi: %d\n", info->dataLength - 1, info->appendedInfo.rssiLatch);
```

Comment:
Quite extensive debugging. Two prints, total of 89 bytes, for every RX event.
In baud rate of 115200 takes about 773 us or more.

Also has potential of dead blocking if normal priority thread have already started to print.